lib/salus/scanners/lang_version.rb

```diff
@@ -0,0 +1,75 @@
+require 'salus/scanners/base'
+require 'json'
+
+module Salus::Scanners
+  class LangVersion < Base
+    class SemVersion < Gem::Version; end
+
+    def run
+      version = SemVersion.new(lang_version)
+      min_version = SemVersion.new(@config['min_version']) if @config['min_version']
+      max_version = SemVersion.new(@config['max_version']) if @config['max_version']
+
+      if min_version && (version < min_version)
+        msg = "Repository language version (#{version}) is less " \
+        "than minimum configured version (#{min_version})"
+        log(msg)
+        return report_failure
+      end
+
+      if max_version && (version > max_version)
+        msg = "Repository language version (#{version}) is greater " \
+        "than maximum configured version (#{max_version})"
+        log(msg)
+        return report_failure
+      end
+
+      report_success
+    end
+
+    def self.supported_languages
+      %w[go ruby]
+    end
+
+    def should_run?
+      go_project? || ruby_project?
+    end
+
+    private
+
+    def lang_version
```

Comment:
We have polyglot projects too, (mono/repo for example has both ruby and golang)

Might be nice to create a simple class for each language

```
#Pseudo code for ruby scanner
class RubyVersionScanner < LanuageVersion
  def lang_version
        cmd = "bundle platform --ruby"
       run_shell(cmd, chdir: @repository.path_to_repo).stdout
      end
end
```

Reply:
@joshua-ostrom: Can you have a look to see this new structure for supporting separate class for each language? Thanks.